### PR TITLE
575 and 78 Cleanup help message

### PR DESCRIPTION
**Pull request by frederikja163**
Cleanup the help message a bit to make it more uniform, update it to use log.info() rather than Console.WriteLine(),
this also means that messages now arrive in the console in the correct order.
Also prints an error message for incorrect command usage.

**Closes** #575 and #78

**Example of message before:**
```
PS C:\Users\fredandr\opentap\bin\Debug> .\tap.exe package notvalid
OpenTAP Command Line Interface (9.18.0-alpha.18+cf121119.i215-UninstallingMultiplePacka)
Usage: tap <command> [<subcommand(s)>] [<args>]

Valid subcommands of
Run "tap.exe <command> [<subcommand>] -h" to get additional help for a specific command.

package
   create                 Create a package based on an XML description file.
   download               Download one or more packages.
   install                Install one or more packages.
   list                   List locally installed packages and browse the online package repository.
   show                   Show information about a package.
   test                   Run tests on one or more packages.
   uninstall              Uninstall one or more packages.
   verify                 Verify the integrity of one or all installed packages by checking their fingerprints.
PS C:\Users\fredandr\opentap\bin\Debug>
```

**Example of message after:**
```
PS C:\Users\fredandr\opentap\bin\Debug> .\tap.exe package notvalid
OpenTAP Command Line Interface (9.18.0-alpha.18+cf121119.i215-UninstallingMultiplePacka)
"tap.exe package notvalid" is not a recognized command.
Usage: "tap.exe <command> [<subcommand(s)>] [<args>]"

Valid subcommands of
package
   create                 Create a package based on an XML description file.
   download               Download one or more packages.
   install                Install one or more packages.
   list                   List locally installed packages and browse the online package repository.
   show                   Show information about a package.
   test                   Run tests on one or more packages.
   uninstall              Uninstall one or more packages.
   verify                 Verify the integrity of one or all installed packages by checking their fingerprints.

Run "tap.exe <command> [<subcommand(s)>] -h" to get additional help for a specific command.

PS C:\Users\fredandr\opentap\bin\Debug>
```

### CODE REVIEW

File: Engine/Cli/CliActionExecutor.cs [hunk 1]
```diff
@@ -269,26 +269,34 @@ string getVersion()
...
                     return TypeData.FromType(typeof(CliActionExecutor)).Assembly.SemanticVersion.ToString(); // OpenTAP is not installed. lets just return this. 
                 }
+                string tapCommand = OperatingSystem.Current == OperatingSystem.Windows ? "tap.exe" : "tap";
 
-                Console.WriteLine("OpenTAP Command Line Interface ({0})", getVersion());
-                Console.WriteLine("Usage: tap <command> [<subcommand(s)>] [<args>]\n");
+                log.Info("OpenTAP Command Line Interface ({0})", getVersion());
+                if (args.Length == 0)
+                {
+                    log.Info($"\"{tapCommand}\" is not a recognized command.");
```
Comment (rmadsen-ks) [inline Engine/Cli/CliActionExecutor.cs:277]: i think we can skip printing this line - if the user just writes "tap.exe" we can assume they just want the help. What do you think?

File: Engine/Cli/CliActionExecutor.cs [hunk 2]
```diff
@@ -269,26 +269,34 @@ string getVersion()
...
+                }
+                else
+                {
+                    log.Info($"\"{tapCommand} {string.Join(" ", args)}\" is not a recognized command.");
```
Comment (rmadsen-ks) [inline Engine/Cli/CliActionExecutor.cs:281]: Maybe this should be log.Error to highlight the problem?

**Review by rmadsen-ks — CHANGES_REQUESTED** [review]
Looks great, just a couple of suggestions.

File: Engine/Cli/CliActionExecutor.cs (hunk 1 above)
Comment (frederikja163) [reply Engine/Cli/CliActionExecutor.cs]: Yeah, this probably makes sense. Change has been made.

File: Engine/Cli/CliActionExecutor.cs (hunk 2 above)
Comment (frederikja163) [reply Engine/Cli/CliActionExecutor.cs]: This is now a log.Error instead.

File: Engine/Cli/CliActionExecutor.cs [hunk 3]
```diff
@@ -269,26 +269,30 @@ string getVersion()
...
                     return TypeData.FromType(typeof(CliActionExecutor)).Assembly.SemanticVersion.ToString(); // OpenTAP is not installed. lets just return this. 
                 }
+                string tapCommand = OperatingSystem.Current == OperatingSystem.Windows ? "tap.exe" : "tap";
 
-                Console.WriteLine("OpenTAP Command Line Interface ({0})", getVersion());
-                Console.WriteLine("Usage: tap <command> [<subcommand(s)>] [<args>]\n");
+                log.Info("OpenTAP Command Line Interface ({0})", getVersion());
+                if (args.Length != 0)
+                {
+                    log.Error($"\"{tapCommand} {string.Join(" ", args)}\" is not a recognized command.");
```
Comment (rmadsen-ks) [inline Engine/Cli/CliActionExecutor.cs:277]: I think we should move the error message up before line 274 "OpenTAP Command Line Interface..." This way it is the first the user sees.

**Review by rmadsen-ks — CHANGES_REQUESTED** [review]
one minor thing, otherwise it looks great.

File: Engine/Cli/CliActionExecutor.cs (hunk 3 above)
Comment (frederikja163) [reply Engine/Cli/CliActionExecutor.cs:276]: I agree, it has been moved up now

**Review by rmadsen-ks — APPROVED** [review]
Looks good!

**Pull request merged**